From: Liam Howlett <liam.howlett@oracle.com>
Subject: mm/mmap: drop VM_BUG_ON() until better tested
Date: Tue, 19 Jul 2022 17:09:00 +0000

The limits that the maple tree return are trimmed so there is enough space.

Link: https://lkml.kernel.org/r/20220719170841.3298839-1-Liam.Howlett@oracle.com
Fixes: Fixes: 06fade172129 ("mm/mmap: use maple tree for unmapped_area{_topdown}")
Signed-off-by: Liam R. Howlett <Liam.Howlett@oracle.com>
Reported-by: Yu Zhao <yuzhao@google.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/mmap.c |    4 ----
 1 file changed, 4 deletions(-)

--- a/mm/mmap.c~mm-mmap-use-maple-tree-for-unmapped_area_topdown-fix
+++ a/mm/mmap.c
@@ -2037,8 +2037,6 @@ static unsigned long unmapped_area(struc
 
 	gap = mas.index;
 	gap += (info->align_offset - gap) & info->align_mask;
-	VM_BUG_ON(gap + info->length > info->high_limit);
-	VM_BUG_ON(gap + info->length > mas.last);
 	return gap;
 }
 
@@ -2068,8 +2066,6 @@ static unsigned long unmapped_area_topdo
 
 	gap = mas.last + 1 - info->length;
 	gap -= (gap - info->align_offset) & info->align_mask;
-	VM_BUG_ON(gap < info->low_limit);
-	VM_BUG_ON(gap < mas.index);
 	return gap;
 }
 
_
